pkg/fastesturl/fastesturl.go

```diff
@@ -55,55 +52,43 @@ func New(client *http.Client, req *http.Request, check RespCheck, urls []*url.UR
 // f.RespCheck.
 //
 // If no successful *http.Response is obtained a nil is returned.
-// Any ctx timeout or cancelation must be provided by the caller.
+// Any timeout or cancellation must be provided by the caller.
 func (f *FastestURL) Do(ctx context.Context) *http.Response {
 	log := zerolog.Ctx(ctx).With().Str("routine", "fasesturl_do").Logger()
-	cond := sync.NewCond(&sync.Mutex{})
-	var response *http.Response
-	// immediately lock so workers do not write to response
-	// before Do method blocks on cond.Wait()
-	cond.L.Lock()
+	var wg sync.WaitGroup
+	result := make(chan *http.Response)
+	wg.Add(len(f.URLs))
+	go func() {
+		wg.Wait()
+		close(result)
+	}()
 	for _, url := range f.URLs {
 		u := url
 		go func() {
-			defer cond.Signal()
```

Comment:
I think under some conditions, this could be hit before the main goroutine was `Wait`ing below, making the tests hang.